angelsrefining/prototypes/angels-functions.lua

```diff
@@ -150,25 +150,116 @@ end
+  U  = {{000, 143, 255}},                                   --Uranium
+  Np = {{000, 128, 255}},                                   --Neptunium
+  Pu = {{000, 107, 255}},                                   --Plutonium
+  Am = {{084, 092, 242}},                                   --Americium
+  Cm = {{120, 092, 227}},                                   --Curium
+  --fake/compicated/custom tints
+  Tw = {{243, 135, 000}},                                   --Thermal water
+  Oi = {{069, 069, 069}, {054, 054, 054}, {036, 036, 036}}, --Coal/Oil
+  Xx = {{041, 041, 180}},                                   --Complex (really strange materials)
+  Ws = {{094, 114, 174}, {088, 104, 163}, {088, 101, 155}}, --Water/Steam
+  Sg = {{255, 105, 180}},                                   --Syngas
+  Ng = {{105, 135, 090}, {096, 122, 082}, {088, 113, 075}}, --Natural Gas
+  Cb = {{015, 015, 015}},                                   --Other Carbon Solids
+  Ax = {{241, 050, 238}},                                   --Alien Stuffs
+  Aw = {{194, 227, 091}, {184, 239, 000}, {156, 207, 000}},  --Alien Feed (gas/water)
 }
+--[[{ unused materials
+  Ne = {{179, 227, 245}}, Ar = {{128, 209, 227}}, Sc = {{230, 230, 230}}, V  = {{166, 166, 171}}, Ga = {{194, 143, 143}}, Ge = {{102, 143, 143}}, As = {{189, 128, 227}}, Se = {{255, 161, 000}}, 
+  Br = {{166, 041, 041}}, Kr = {{092, 184, 209}}, Rb = {{112, 046, 176}}, Sr = {{000, 255, 000}}, Y  = {{148, 255, 255}}, Zr = {{148, 224, 224}}, Nb = {{115, 194, 201}}, Mo = {{084, 181, 181}}, 
+  Tc = {{059, 158, 158}}, Ru = {{036, 143, 143}}, Rh = {{010, 125, 140}}, Pd = {{000, 105, 133}}, Cd = {{255, 217, 143}}, In = {{166, 117, 115}}, Sb = {{158, 099, 181}}, Te = {{212, 122, 000}}, 
+  I  = {{148, 000, 148}}, Xe = {{066, 158, 176}}, Ba = {{000, 201, 000}}, La = {{112, 212, 255}}, Ce = {{255, 255, 199}}, Pr = {{217, 255, 199}}, Nd = {{199, 255, 199}}, Pm = {{163, 255, 199}}, 
+  Sm = {{143, 255, 199}}, Eu = {{097, 255, 199}}, Gd = {{069, 255, 199}}, Tb = {{048, 255, 199}}, Dy = {{031, 255, 199}}, Ho = {{000, 255, 156}}, Er = {{000, 230, 117}}, Tm = {{000, 212, 082}},  
+  Yb = {{000, 191, 056}}, Lu = {{000, 171, 036}}, Hf = {{077, 194, 255}}, Ta = {{077, 166, 255}}, Re = {{038, 125, 171}}, Ir = {{023, 084, 135}}, Tl = {{166, 084, 077}}, Bi = {{158, 079, 181}}, 
+  At = {{117, 079, 069}}, Rn = {{066, 130, 150}}, Fr = {{066, 000, 102}}, Ra = {{000, 125, 000}}, Ac = {{112, 171, 250}}, Bk = {{138, 079, 227}}, Cf = {{161, 054, 212}}, Es = {{179, 031, 212}}, 
+  Fm = {{179, 031, 186}}, Md = {{179, 013, 166}}, No = {{189, 013, 135}}, Lr = {{199, 000, 102}}, Rf = {{204, 000, 089}}, Db = {{209, 000, 079}}, Sg = {{217, 000, 069}}, Bh = {{224, 000, 056}}, 
+  Hs = {{230, 000, 046}}, Mt = {{235, 000, 038}}, }]]
+
+local function get_molecule_codes(molec_formula)
+  local string_codes = {} 
+  while string.len(molec_formula) > 0 do
+  local trim = 1
+    if string.find(molec_formula,"(%u%l)%d+") == 1 then --do it bit-wise

```

Comment:
```suggestion
  while string.len(molec_formula) > 0 do
    local trim = 1
    if string.find(molec_formula,"(%u%l)%d+") == 1 then --do it bit-wise
```

Reply:
apart from the spacing issue, do you have any problems with that function?